src/components/FlawHistory.vue

```diff
@@ -0,0 +1,55 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+
+import { DateTime } from 'luxon';
+
+import { capitalizeFirstLetter } from '@/utils/helpers';
+import { flawFieldNamesMapping } from '@/constants/flawFields';
+import type { ZodFlawType } from '@/types/zodFlaw';
+
+import LabelCollapsible from './widgets/LabelCollapsible.vue';
+
+defineProps<{
+  flaw: ZodFlawType;
+}>();
+
+const historyExpanded = ref(false);
+</script>
+
+<template>
+  <LabelCollapsible
+    v-show="flaw.history && flaw.history.length > 0"
+    class="my-2"
+    :isExpanded="historyExpanded"
+    @toggleExpanded="historyExpanded = !historyExpanded"
+  >
+    <template #label>
+      <label class="mx-2 mb-0 form-label">
+        History:
+      </label>
+    </template>
+    <div class="mt-2">
+      <label class="mx-2 form-label w-100">
+        <template v-for="(historyEntry, histIndex) in flaw.history" :key="histIndex">
+          <div v-if="historyEntry.pgh_diff" class="alert alert-info mb-1 p-2">
+            <span>
+              {{ DateTime.fromISO(historyEntry.pgh_created_at || '',{ setZone: true })
+                .toFormat('yyyy-MM-dd hh:mm ZZZZ') }} - {{ historyEntry.pgh_context?.user || 'System' }}
+            </span>
+            <ul class="mb-2">
+              <li v-for="(diffEntry, diffKey, diffIndex) in historyEntry.pgh_diff" :key="diffIndex">
```

Comment:
Again, if the data has a key I would prefer to use it instead of the index
```suggestion
              <li v-for="(diffEntry, diffKey)  in historyEntry.pgh_diff" :key="diffKey">
```